spec/controllers/raw_file_sources_controller_spec.rb

```diff
@@ -0,0 +1,202 @@
+require 'rails_helper'
+
+require 'support/controller_macros'
+require 'support/devise'
+require 'support/shared_examples_for_authentication'
+
+RSpec.describe RawFileSourcesController, type: :controller do
+	it_behaves_like "an authenticating controller", :index, "raw_file_sources"
+	
+	describe "GET index" do
+		login_user
+		render_views
+
+		before(:each) do
+			@rfs = create :raw_file_source
+			get :index
+		end
+
+		it "populates an array of sources" do
+			expect(assigns(:raw_file_sources)).to eq([@rfs])
+		end
+
+		it "renders the index page" do
+			expect(response).to render_template(:index)
+			expect(response.content_type).to eq("text/html")
+		end
+
+		it "displays raw file sources" do
+			expect(response.body).to match /file_source_/im
+		end
+	end
+
+	describe "GET new" do
+		login_user
+		render_views
+
+		before(:each) do
+			create :raw_file_source
+			get :new
+		end
+
+		it "renders the new page" do
```

Comment:
Does this test just check to see if the response type is HTML? Does it need to be checking other things to verify that data is being pulled correctly?